examples/tutorials/tableloader_and_eventsources.py

```diff
@@ -0,0 +1,286 @@
+maxlen = max(len(source) for source in sources)
+for source in sources.values():
+    print(f"{source.__name__:>{maxlen}s} -- {source.__module__}.{source.__qualname__}")
+
+# %%
+# ``EventSource`` will create an appropriate subclass if a compatible source is found for
+# the given ``input_url`` :
+
+source = EventSource(input_url=simtel_path, max_events=5)
+print(source)
+
+# %%
+# You can now loop over the ``ctapipe.containers.ArrayEventContainer`` generated by the
+# source.
+
+for event in source:
+    print(event.count)
+
+# %%
+
+print(repr(event))
+
+# %%
+# Every time a new loop is started through the source,
+# it tries to restart from the first event, which might not be supported
+# by the event source. It is encouraged to use ``EventSource`` in a **context manager**
+# to ensure the correct cleanups are performed when you are finished with the source:
+
+with EventSource(input_url=simtel_path) as source:
+    for event in source:
+        print(
+            f"Event Count: {event.count},"
+            f"Tels with trigger: {event.trigger.tels_with_trigger}"
+        )
+
+# %%
+# You can hand in the ID's of the telescopes to be included in the data
+# with the ``allowed_tels`` attribute. If given, only this subset of telescopes
+# will be present in the generated events. If None, all available telescopes are used.
+
+eventsource_config = Config(
+    {"EventSource": {"max_events": 5, "allowed_tels": [3, 4, 9]}}
+)
+
+with EventSource(input_url=simtel_path, config=eventsource_config) as source:
+    for event in source:
+        print(
+            f"Event Count: {event.count},"
+            f"Tels with trigger: {event.trigger.tels_with_trigger}"
+        )
+
+# %%
+# If you want to calibrate your data in the event loop and write it to an .h5 file with
+# the ``DataWriter`` :
+#
+
+source = EventSource(input_url=simtel_path, max_events=50)
+calibrate = CameraCalibrator(subarray=source.subarray)
+
+with DataWriter(
+    event_source=source,
+    output_path="events.dl1.h5",
+    write_parameters=False,
+    overwrite=True,
+    write_images=True,
+) as write_data:
+    for event in source:
+        calibrate(event)
+        write_data(event)
+
+# %%
+# Alternatively doing it with ``ctapipe-process`` would look like this:
+#
+# .. code-block:: bash
+#
+#   ! ctapipe-process -i {simtel_path} -o events.dl1.h5 --overwrite --progress
+#
+
+# %%
+# TableLoader
+# -----------
+#
+# Create a TableLoader instance with the above created dl1 file:
+
+loader = TableLoader(input_url="events.dl1.h5")
+
+# %%
+# Alternatively using a config file:
+
+tableloader_config = Config(
+    {
+        "TableLoader": {
+            "input_url": "events.dl1.h5",
+        }
+    }
+)
+
+loader = TableLoader(config=tableloader_config)
+
+# %%
+# Reading subarray-based event information:
+
+subarray_events = loader.read_subarray_events(
+    start=None,
+    stop=None,
+    dl2=False,
+    simulated=True,
+    observation_info=False,
+)
+
+print(subarray_events)
```

Comment:
I suggest either just use `subarray_events` (no print()) or alternatively import the display() function from Ipython and use that. Otherwise the documentation looks ugly, as the tables are rendered in text and not withj the HTML repr:
![image](https://github.com/cta-observatory/ctapipe/assets/11677812/84b03559-06a3-4066-b4e5-74354b1dbfc9)